src/Handler/Bucket/UnShare/UnShareBucketHandler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\BigQuery\Handler\Bucket\UnShare;
+
+use Google\Protobuf\Internal\Message;
+use Keboola\StorageDriver\BigQuery\GCPClientManager;
+use Keboola\StorageDriver\Command\Bucket\UnshareBucketCommand;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+
+final class UnShareBucketHandler implements DriverCommandHandlerInterface
+{
+    public GCPClientManager $clientManager;
+
+    public function __construct(GCPClientManager $clientManager)
+    {
+        $this->clientManager = $clientManager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     */
+    public function __invoke(
+        Message $credentials, // backend credentials
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof UnshareBucketCommand);
+
+        assert($command->getBucketShareRoleName() !== '');
+
+        $analyticHubClient = $this->clientManager->getAnalyticHubClient($credentials);
+
+        $analyticHubClient->deleteListing($command->getBucketShareRoleName());
```

Comment:
Bohizial zmazanie listingu nezmaze nalinkovane buckety, slo by to este osterit tak ze si vylistujem pomocou query vsetky linknute miesta a druhou query zmazem ale nieje to atomicke takze by tam mohlo ostat nieco vyset. 

Je to ale na strane KBC osetrene ze nejde dat unshare na bucket ktory je niekde nalinkovany